WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -590,11 +541,73 @@ - (void)setEditing:(BOOL)editing animated:(BOOL)animated
     }
 }
 
+- (void)toggleRightBarButtonItems:(BOOL)enabled
+{
+    for (UIBarButtonItem *buttonItem in self.navigationItem.rightBarButtonItems) {
+        buttonItem.enabled = enabled;
+    }
+}
+
 - (void)toggleSearch
 {
     self.searchController.active = !self.searchController.active;
 }
 
+- (void)addSite
+{
+    UIAlertController *addSiteAlertController = [UIAlertController alertControllerWithTitle:nil
+                                                                                    message:nil
+                                                                             preferredStyle:UIAlertControllerStyleActionSheet];
+    UIAlertAction *addNewWordPressAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Create WordPress.com site", @"Create WordPress.com site button")
+                                                                    style:UIAlertActionStyleDefault
+                                                                  handler:^(UIAlertAction *action) {
+                                                                      [self showAddNewWordPressController];
+                                                                  }];
+    UIAlertAction *addSiteAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Add self-hosted site", @"Add self-hosted site button")
+                                                            style:UIAlertActionStyleDefault
+                                                          handler:^(UIAlertAction *action) {
+                                                              [self showLoginControllerForAddingSelfHostedSite];
+                                                          }];
+    UIAlertAction *cancel = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", @"Cancel button")
+                                                     style:UIAlertActionStyleCancel
+                                                   handler:nil];
+    
+    [addSiteAlertController addAction:addNewWordPressAction];
+    [addSiteAlertController addAction:addSiteAction];
+    [addSiteAlertController addAction:cancel];
+    addSiteAlertController.popoverPresentationController.barButtonItem = self.navigationItem.rightBarButtonItem;
```

Comment:
@kurzee rightBarbuttonItem is the same as rightBarButtonItem[0] according to the apple docs: https://developer.apple.com/library/ios/documentation/UIKit/Reference/UINavigationItem_Class/index.html

> If there is not enough room to display all of the items in the array, those that would overlap the title view (if present) or the buttons on the left side of the bar are not displayed.

>The first item in the array can also be set using the rightBarButtonItem property.

Because of this I am comfortable letting it stay the same. However if worries you, I'll change it.